### PR TITLE
Convert2RHEL - support for Oracle 7

### PR DESCRIPTION
**Pull request by stejskalleos**
Followup to [PR Convert2RHEL role](https://github.com/theforeman/foreman-ansible-modules/pull/1286)

### CODE REVIEW

**Comment by stejskalleos** [comment]
@ares @evgeni

File: roles/convert2rhel/tasks/products_and_repos.yml [hunk 1]
```diff
@@ -52,6 +52,24 @@
             state: present
   when: foreman_content_rhel_enable_rhel8
 
+- name: "Create 'UBI' product and repositories"
```
Comment (evgeni) [inline roles/convert2rhel/tasks/products_and_repos.yml]: what is this repo and why is this needed? (a doc link to make my brain happy is sufficient)

File: roles/convert2rhel/defaults/main.yml [hunk 2]
```diff
@@ -16,3 +17,7 @@ foreman_convert2rhel_content_view: "Default Organization View"
 # Host groups
 foreman_convert2rhel_hostgroup7: "CentOS 7 converting"
 foreman_convert2rhel_hostgroup8: "CentOS 8 converting"
+foreman_convert2rhel_hg_ol7: "Oracle 7 converting"
```
Comment (evgeni) [inline roles/convert2rhel/defaults/main.yml]: should this be `foreman_convert2rhel_hostgroup_ol7` to match the others?

or should the others also end in `_centosX`?

File: roles/convert2rhel/tasks/products_and_repos.yml (hunk 1 above)
Comment (ares) [reply roles/convert2rhel/tasks/products_and_repos.yml]: OL does not have subscription-manager available in its repositories, but we need that for the registration, the best source for that is the [UBI](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image) repository. Convert2rhel installs the subscription-manager from there when it's not used with Satellite (and expects the Internet connectivity)

**Comment by evgeni** [comment]
#1286 is in, this now needs a rebase and the comments adressed

File: roles/convert2rhel/tasks/products_and_repos.yml (hunk 1 above)
Comment (evgeni) [reply roles/convert2rhel/tasks/products_and_repos.yml]: I think we should call the product/repos differently. While they come from UBI, this is not what we use them for and this will lead to confusion.

How about "Oracle Linux subscription-manager dependencies" or something? Or "Oracle Linux convert2rhel dependencies"?

**Comment by stejskalleos** [comment]
@evgeni rebased and updated

I will update tests later when I solve the issue with oracle repo sync:
```
Cannot connect to host cdn-ubi.redhat.com:443 ssl:default [Network is unreachable]
```

File: roles/convert2rhel/README.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ This role supports the [Common Role Variables](https://github.com/theforeman/for
 - `foreman_convert2rhel_manage_subscription`: Run [content_rhel Role](https://github.com/theforeman/foreman-ansible-modules/blob/develop/roles/content_rhel/README.md) role, default: `true`
 - `foreman_convert2rhel_lifecycle_env`: Lifecycle environment for activation keys, default: Library.
 - `foreman_convert2rhel_content_view`: Content view for activation keys, default: Default Organization View.
+- `foreman_convert2rhel_enable_oracle7`: Create data for Oracle 7 conversion, default: `false`
```
Comment (evgeni) [inline roles/convert2rhel/README.md]:
```suggestion
- `foreman_convert2rhel_enable_oracle7`: Create data for Oracle Linux 7 conversion, default: `false`
```

File: roles/convert2rhel/defaults/main.yml [hunk 4]
```diff
@@ -18,3 +21,7 @@ foreman_convert2rhel_content_view: "Default Organization View"
 # Host groups
 foreman_convert2rhel_hostgroup7: "CentOS 7 converting"
 foreman_convert2rhel_hostgroup8: "CentOS 8 converting"
+foreman_convert2rhel_hg_oracle7: "Oracle 7 converting"
```
Comment (evgeni) [inline roles/convert2rhel/defaults/main.yml]: I still think that the hostgroup variables should have identical structure. And my vote would be:
* foreman_convert2rhel_hostgroup_centos7
* foreman_convert2rhel_hostgroup_centos8
* foreman_convert2rhel_hostgroup_oracle7

And you're converting Oracle *Linux*, so while I think it's fine to shorten it in the variable name, the HG name should be "Oracle Linux 7 converting"

**Review by evgeni — CHANGES_REQUESTED** [review]
two minor nitpicks, LGTM otherwise and can be merged once nitpicks are adressed.

**Review by evgeni — APPROVED** [review]
